src/main/scala/io/iohk/ethereum/domain/BlockHeader.scala

```diff
@@ -40,3 +43,14 @@ case class BlockHeader(
          |}""".stripMargin
     }
   }
+
+object BlockHeader {
+    /**
+      * calculates blockHash for given block header
+      * @param header - block header for which hash will be calculated
+      * @return - hash that can be used to get block bodies / receipts
+      */
+    def hash(header: BlockHeader): ByteString = {
```

Comment:
I was talking with @whysoserious and he is suggesting that we keep domain as pure case classes, but we can talk about that